distutils/command/build_ext.py

```diff
@@ -125,7 +125,7 @@ def initialize_options(self):
         self.rpath = None
         self.link_objects = None
         self.debug = None
-        self.force = None
+        self.force: bool = None  # Should always be set in finalize_options
```

Comment:
Not actually type-safe and easy to break in the future, but this is only place in distutils we assume a type for after `finalize_options`.

Alternatively this can be kept as  `bool | None`, which still works everywhere expecting a falsy/truthy value. But in some places where an actual boolean is expected (like https://github.com/pypa/setuptools/blob/d198e86f57231e83de87975c5c82bc40c196da79/setuptools/command/build_ext.py#L225-L227 ) we'd have to coerce to a boolean: `bool(self.force)`